src/odemis/driver/lakeshore.py

```diff
@@ -198,6 +195,7 @@ def _findDevice(self, ports):
                     continue
 
                 self._serial = self._openSerialPort(n)
+                self.GetIdentifier()  # if value is incorrect, will throw an exception wile unpacking
```

Comment:
That's no enough, because it's going to raise a TypeError, which you don't handle.
I think that the neatest would be to handle such TypeError inside GetIdentifier (when it's due to no having enough commas in the returned string), and raise IOError in such case.

Also, this is technically just checking that the identifier string returns at least 3 commas, it's not so specific to this device. Please also check the the `manufacter.lower() == "lakeshore"`.

Reply:
Should be fixed now.